src/community/wps-download-netcdf/src/test/java/org/geoserver/wps/ppio/NetCDFDownloadProcessTest.java

```diff
@@ -0,0 +1,222 @@
+/* (c) 2025 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wps.ppio;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.awt.image.RenderedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+import javax.xml.namespace.QName;
+import org.geoserver.catalog.DimensionPresentation;
+import org.geoserver.catalog.ResourceInfo;
+import org.geoserver.config.GeoServer;
+import org.geoserver.data.test.SystemTestData;
+import org.geoserver.platform.GeoServerExtensions;
+import org.geoserver.wcs.responses.AbstractNetCDFEncoder;
+import org.geoserver.wps.WPSTestSupport;
+import org.geoserver.wps.gs.download.DownloadEstimatorProcess;
+import org.geoserver.wps.gs.download.DownloadProcess;
+import org.geoserver.wps.gs.download.DownloadProcessTest;
+import org.geoserver.wps.gs.download.Parameter;
+import org.geoserver.wps.gs.download.Parameters;
+import org.geoserver.wps.gs.download.StaticDownloadServiceConfiguration;
+import org.geoserver.wps.process.RawData;
+import org.geoserver.wps.process.ResourceRawData;
+import org.geoserver.wps.resource.WPSResourceManager;
+import org.geotools.api.data.DataSourceException;
+import org.geotools.api.filter.Filter;
+import org.geotools.api.filter.FilterFactory;
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.geotools.coverage.io.netcdf.NetCDFReader;
+import org.geotools.coverage.util.FeatureUtilities;
+import org.geotools.data.util.NullProgressListener;
+import org.junit.Test;
+import ucar.ma2.DataType;
+import ucar.nc2.Attribute;
+import ucar.nc2.Dimension;
+import ucar.nc2.Variable;
+import ucar.nc2.dataset.DatasetUrl;
+import ucar.nc2.dataset.NetcdfDataset;
+import ucar.nc2.dataset.NetcdfDatasets;
+
+public class NetCDFDownloadProcessTest extends WPSTestSupport {
+
+    private static QName TIMESERIES = new QName(WCS_URI, "timeseries", WCS_PREFIX);
+    private static final FilterFactory FF = FeatureUtilities.DEFAULT_FILTER_FACTORY;
+
+    @Override
+    protected void onSetUp(SystemTestData testData) throws Exception {
+        super.onSetUp(testData);
+        testData.addRasterLayer(TIMESERIES, "timeseries.zip", null, getCatalog());
+        setupRasterDimension(TIMESERIES, ResourceInfo.TIME, DimensionPresentation.LIST, null, "ISO8601", null);
+    }
+
+    protected WPSResourceManager getResourceManager() {
+        return GeoServerExtensions.bean(WPSResourceManager.class);
+    }
+
+    public static class AutoDisposableNetCDFReader extends NetCDFReader implements AutoCloseable {
```

Comment:
One day we should make  all readers AutoCloseable. Something to consider as a stretch goal for GeoServer 3 I guess.